0000-nonlexical-lifetimes.md

```diff
@@ -1575,8 +1579,8 @@ Another example, this time using a `match`:
 ```rust
 fn main() {
     let mut x = Some(3);
-    match &mut x {
-        Some(i) => {
+    match *&mut x {
+        Some(ref mut i) => {
```

Comment:
I can't remember, but I think I may have been intending to use the notation introduced by https://github.com/rust-lang/rfcs/pull/2005 here (which is not, admittedly, yet implemented).